crates/signer/src/manager.rs

```diff
@@ -180,20 +183,38 @@ impl SigningManager {
         }
     }
 
-    pub fn get_delegation(
+    // The trait bound is merely an implementational detail, we don't want this
+    // trait to be implemented outside.
+    #[allow(private_bounds)]
+    pub fn get_delegation<T>(
         &self,
-        pubkey: &GenericPubkey,
-    ) -> Result<SignedProxyDelegation, SignerModuleError> {
-        let delegation: Option<SignedProxyDelegation> = match pubkey {
-            GenericPubkey::Bls(bls_pk) => {
-                self.proxy_signers.bls_signers.get(bls_pk).map(|x| x.delegation).map(Into::into)
-            }
-            GenericPubkey::Ecdsa(ecdsa_pk) => {
-                self.proxy_signers.ecdsa_signers.get(ecdsa_pk).map(|x| x.delegation).map(Into::into)
-            }
-        };
+        pubkey: &T,
+    ) -> Result<SignedProxyDelegation<T>, SignerModuleError>
+    where
+        T: PublicKey,
+        Self: GetDelegation<T>,
+    {
+        <Self as GetDelegation<T>>::get_delegation(self, pubkey)
+            .ok_or(SignerModuleError::UnknownProxySigner(pubkey.as_ref().to_vec()))
+    }
+}
+
+trait GetDelegation<T: PublicKey> {
+    fn get_delegation(&self, pubkey: &T) -> Option<SignedProxyDelegation<T>>;
+}
+
+impl GetDelegation<BlsPublicKey> for SigningManager {
+    fn get_delegation(&self, pubkey: &BlsPublicKey) -> Option<SignedProxyDelegation<BlsPublicKey>> {
+        self.proxy_signers.bls_signers.get(pubkey).map(|x| x.delegation)
+    }
+}
 
-        delegation.ok_or(SignerModuleError::UnknownProxySigner(pubkey.as_ref().to_vec()))
+impl GetDelegation<EcdsaPublicKey> for SigningManager {
+    fn get_delegation(
+        &self,
+        pubkey: &EcdsaPublicKey,
+    ) -> Option<SignedProxyDelegation<EcdsaPublicKey>> {
+        self.proxy_signers.ecdsa_signers.get(pubkey).map(|x| x.delegation)
```

Comment:
That's the generic implementation of `get_delegation`. 
If we deem this unwanted as too complex, then we can either:
1) [the simplest] get rid of this method as it's still unused anywhere
2) go the other way and duplicate two methods: `get_delegation_ecdsa` and `get_delegation_bls`
3) go full circle and make the proxy signers, the signers, etc. also generic (this time with a trait bound to the public key instead of the secret keys), but this is going to infest the signer module with generics overall again. :D

Reply:
Agreed. Refactored in [7ddcd3a](https://github.com/Commit-Boost/commit-boost-client/pull/87/commits/7ddcd3a67764367d26efac87f59509f3bc437505)